src/Arbor/File/Format/Asif/Format/Decoder.hs

```diff
@@ -0,0 +1,48 @@
+module Arbor.File.Format.Asif.Format.Decoder where
+
+import Arbor.File.Format.Asif.Data.Ip
+import Control.Lens
+import Data.Thyme.Time.Core
+import Data.Word                             (Word32)
+import HaskellWorks.Data.Network.Ip.Validity (Canonical)
+
+import qualified Data.Binary.Get                   as G
+import qualified HaskellWorks.Data.Network.Ip.Ipv4 as IP4
+import qualified HaskellWorks.Data.Network.Ip.Ipv6 as IP6
+
+getWord32x4 :: G.Get (Word32, Word32, Word32, Word32)
+getWord32x4 = do
+  a <- G.getWord32be
+  b <- G.getWord32be
+  c <- G.getWord32be
+  d <- G.getWord32be
```

Comment:
Other cases use `getWord32le`, are you use `getWord32be` here deliberately?

Reply:
It's deliberate in the sense that Dave said they did it deliberately.

Reply:
Probably because `IPv6` is stored in Network Order, which is the same as BE.

Reply:
It's interesting though that we store IPv4 in LE, so there is some inconsistency there.